#161342791 Setup an empty react project and host it on heroku

ucheg6 opened the pull request:
#### What does this PR do?
Setup an empty react project
#### Description of Task to be completed?
- setup and configure webpack
- configure babel to transpile to es5
- configure eslint with airbnb styleguide

#### How should this be manually tested?
- Git clone the repository
- cd into the `haven-ah-frontend` folder
- run `npm run start:dev`
- To see the production build visit [`Authors Haven App`](https://haven-ah-frontend.herokuapp.com/)

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#161342791](https://www.pivotaltracker.com/story/show/161342791)
#### Screenshots (if appropriate)
<img width="1439" alt="screen shot 2018-11-26 at 3 25 18 pm" src="https://user-images.githubusercontent.com/34472497/49019974-92c80700-f18f-11e8-9551-c7f02485f009.png">
<img width="1440" alt="screen shot 2018-11-26 at 4 50 11 pm" src="https://user-images.githubusercontent.com/41582565/49025284-96fa2180-f19b-11e8-8ab2-617dd9f10607.png">


#### Questions:
N/A

seyi-adeleke commented on webpack.config.js:
```diff
@@ -0,0 +1,42 @@
+        loader: 'file-loader?name=images/[name].[ext]',
+      },
+      {
+        test: /\.js$/,
```
please remove this

seyi-adeleke commented on package.json:
```diff
@@ -0,0 +1,47 @@
+    "eslint-plugin-import": "^2.14.0",
+    "eslint-plugin-jsx-a11y": "^6.1.2",
+    "eslint-plugin-react": "^7.11.1",
+    "file-loader": "^2.0.0",
```
file-loader is unused

ucheg6 replied:
This is being used to load images

seyi-adeleke replied:
At this very moment is this being
 used?

seyi-adeleke reviewed (changes requested):
...

seyi-adeleke commented on webpack.config.prod.js:
```diff
@@ -0,0 +1,11 @@
+
+module.exports = merge(common, {
+  mode: 'production',
+  devtool: 'source-map',
```
we dont need source maps on prod

victor-nweze-deel committed:
ch(react setup): Setup an empty react project and host it on heroku

setup and configure webpack
configure babel to transpile to es5
configure eslint with airbnb styleguide
